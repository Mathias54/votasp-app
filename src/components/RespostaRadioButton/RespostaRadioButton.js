import React from 'react';

import './RespostaRadioButton.css';

const RespostaRadioButton = ({ label, value, id, onClick, checked }) => {
  return (
    <div className="resposta-radio-btn__container">
      <input
        id={id}
        type="radio"
        name="resposta-radio-btn"
        className="resposta-radio-btn__input"
        value={value}
        onClick={onClick}
        checked={checked}
      />
      <span className="resposta-radio-btn__icon" />
      <label htmlFor={id} className="resposta-radio-btn__label">{label}</label>
    </div>
  );
};

export default RespostaRadioButton;
