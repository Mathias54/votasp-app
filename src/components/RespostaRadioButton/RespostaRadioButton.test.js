import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RespostaRadioButton from './RespostaRadioButton';

const defaultProps = {
  label: 'foo',
  value: 'bar',
  id: '1',
  checked: false,
};

describe('<RespostaRadioButton />', () => {
  it('should match snapshot', () => {
    const respostaRadioButton = shallow(<RespostaRadioButton {...defaultProps} />);

    expect(toJson(respostaRadioButton)).toMatchSnapshot();
  });

  it('should call onClick passed function', () => {
    const onClick = jest.fn();
    const respostaRadioButton = shallow((
      <RespostaRadioButton
        {...defaultProps}
        onClick={onClick}
      />
    ));

    respostaRadioButton.find('.resposta-radio-btn__input').simulate('click');

    expect(onClick).toBeCalled();
  });
});
