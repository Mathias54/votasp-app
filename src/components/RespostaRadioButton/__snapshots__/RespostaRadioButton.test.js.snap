// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RespostaRadioButton /> should match snapshot 1`] = `
<div
  className="resposta-radio-btn__container"
>
  <input
    checked={false}
    className="resposta-radio-btn__input"
    id="1"
    name="resposta-radio-btn"
    type="radio"
    value="bar"
  />
  <span
    className="resposta-radio-btn__icon"
  />
  <label
    className="resposta-radio-btn__label"
    htmlFor="1"
  >
    foo
  </label>
</div>
`;
